src/test/ui/issues/issue-72181-1.rs

```diff
@@ -0,0 +1,20 @@
+// build-pass
+// compile-flags: --emit=mir -Z mir-opt-level=1
```

Comment:
Please make this test a `mir-opt` test and emit the MIR of `main` and `f`

Reply:
Fixed!